src/charm.py

```diff
@@ -307,6 +313,55 @@ def check_pgb_running(self):
 
         return True
 
+    def get_hostname_by_unit(self, unit_name: str) -> str:
+        """Create a DNS name for a PgBouncer unit.
+
+        Args:
+            unit_name: the juju unit name, e.g. "pgbouncer-k8s/1".
+
+        Returns:
+            A string representing the hostname of the PgBouncer unit.
+        """
+        unit_id = unit_name.split("/")[1]
+        return f"{self.app.name}-{unit_id}.{self.app.name}-endpoints"
+
+    def get_secret(self, scope: str, key: str) -> Optional[str]:
+        """Get secret from the secret storage."""
+        return self.peers.get_secret(scope, key)
+
+    def set_secret(self, scope: str, key: str, value: Optional[str]) -> None:
+        """Set secret from the secret storage."""
+        self.peers.set_secret(scope, key, value)
+
+    def push_tls_files_to_workload(self) -> None:
+        """Uploads TLS files to the workload container."""
+        key, ca, cert = self.tls.get_tls_files()
+        if key is not None:
+            self.push_file(
+                f"{PGB_DIR}/{TLS_KEY_FILE}",
+                key,
+                0o400,
+            )
```

Comment:
Since the tls lib isn't using the container parameter, I've decided to use the preexisting push file  method